### PR TITLE
ci: Use GITHUB_TOKEN instead of PAT in generate_content.yaml

### PR DESCRIPTION
**Pull request by tungbq**
Based on feedback from @cicirello in this blog post on DEV.TO: https://dev.to/tungbq/github-create-your-own-repository-landscape-16i5
Comment: https://dev.to/cicirello/comment/2ekfe
> We don't actually need a PAT to create a PR from a GitHub Actions workflow. The GITHUB_TOKEN is sufficient for PRs, with the right permissio

This PR is to update the CI workflow to use `GITHUB_TOKEN` instead of `PAT` in generate_content.yaml

Changes:
- .github/workflows/generate_content.yaml to use GITHUB_TOKEN
- Update document for create-repo-landscape.md accordingly

Test OK: https://github.com/tungbq/repos/actions/runs/8946629840

Thank you @cicirello and @ccoVeille for your feedback and suggestion in my blogpost!

### CODE REVIEW

**Comment by ccoVeille** [comment]
My pleasure

**Comment by cicirello** [comment]
You're welcome

**Comment by TWiStErRob** [comment]
You can take it one more step further: when you created the PAT in your blog post, you said only give it `repo` for security, but here you ask people to enable ALL READ/WRITE permissions on their repo. This is not necessary, even with the second (safer) option, you can ask the workflow to get a GITHUB_TOKEN with the right permissions. For more info: https://ghlint.twisterrob.net/issues/default/MissingJobPermissions/

Long story short, add this to the job:
```yml
permissions:
  contents: write # For pushing the branch.
  pull-requests: write # For creating the PR.
```
and remove "Read and write permissions" from the setup steps.

(Disclaimer: I do way too much GHA in my free time and at work and I wrote ghlint to encode some of my learnings in automation.)

**Comment by tungbq** [comment]
Thank you @TWiStErRob for your feedback. Seems like it brings a better security control, let me check and adjust my CI workflow to improve my repository.
